publication/src/main/scala/hmda/publication/reports/util/PricingDataUtil.scala

```diff
@@ -0,0 +1,127 @@
+      rs2_0 <- pricingDisposition(lars, rateSpreadBetween(2, 2.5), "2.00 - 2.49")
+      rs2_5 <- pricingDisposition(lars, rateSpreadBetween(2.5, 3), "2.50 - 2.99")
+      rs3 <- pricingDisposition(lars, rateSpreadBetween(3, 4), "3.00 - 3.99")
+      rs4 <- pricingDisposition(lars, rateSpreadBetween(4, 5), "4.00 - 4.99")
+      rs5 <- pricingDisposition(lars, rateSpreadBetween(5, Int.MaxValue), "5 or more")
+      mean <- reportedMean(lars)
+      median <- reportedMedian(lars)
+    } yield {
+      s"""
+         |[
+         |    $noData,
+         |    $reported,
+         |    $rs1_5,
+         |    $rs2_0,
+         |    $rs2_5,
+         |    $rs3,
+         |    $rs4,
+         |    $rs5,
+         |    $mean,
+         |    $median
+         |]
+     """.stripMargin
+    }
+  }
+
+  private def rateSpreadBetween(lower: Double, upper: Double)(lar: LoanApplicationRegister): Boolean = {
+    Try(lar.rateSpread.toDouble) match {
+      case Success(value) => value >= lower && value < upper
+      case _ => false
+    }
+  }
+
+  private def pricingDataReported(lar: LoanApplicationRegister): Boolean = {
+    rateSpreadBetween(Int.MinValue, Int.MaxValue)(lar)
+  }
+
+  private def pricingDisposition[ec: EC, mat: MAT, as: AS](larSource: Source[LoanApplicationRegister, NotUsed], filter: LoanApplicationRegister => Boolean, title: String): Future[String] = {
+    val loansFiltered = larSource.filter(filter)
+    val loanCountF = count(loansFiltered)
+    val valueSumF = sum(loansFiltered, loanAmount)
+    for {
+      count <- loanCountF
+      totalValue <- valueSumF
+    } yield {
+      s"""
+         |{
+         |    "pricing": "$title",
+         |    "count": $count,
+         |    "value": $totalValue
+         |}
+       """.stripMargin
+    }
+  }
+
+  private def loanAmount(lar: LoanApplicationRegister): Int = lar.loan.amount
+  private def rateSpread(lar: LoanApplicationRegister): Double =
+    Try(lar.rateSpread.toDouble).getOrElse(0)
+
+  private def reportedMean[ec: EC, mat: MAT, as: AS](lars: Source[LoanApplicationRegister, NotUsed]): Future[String] = {
+    val loansFiltered = lars.filter(pricingDataReported)
+    val loanCountF = count(loansFiltered)
+    val rateSpreadSumF = sumDouble(loansFiltered, rateSpread)
+    for {
+      count <- loanCountF
+      totalRateSpread <- rateSpreadSumF
+    } yield {
+      val mean = if (count == 0) "\"\""
+      else {
+        val v = totalRateSpread / count
+        BigDecimal(v).setScale(2, BigDecimal.RoundingMode.HALF_UP).toDouble
+      }
+
+      s"""
+         |{
+         |    "pricing": "Mean",
+         |    "count": $mean,
+         |    "value": "None"
+         |}
+       """.stripMargin
+    }
+  }
+
+  private def reportedMedian[ec: EC, mat: MAT, as: AS](lars: Source[LoanApplicationRegister, NotUsed]): Future[String] = {
+    val rateSpreadsF: Future[Seq[Double]] =
+      lars.filter(pricingDataReported)
+        // .limit(MAX_SIZE)
```

Comment:
Is this needed at all?  Just want to make sure it's meant to be commented out.

Reply:
Oh yeah. I've been thinking about that (thanks for the reminder). I think we should leave it commented unless we experience issues with that method.

It exists to make sure that converting a `Source` to a `Seq` doesn't overflow app memory, but the sets of LARs that actually get sent to that method are going to be pretty small. If anyone feels strongly that we should have it, I'll have to do a little research into what value to set `MAX_SIZE` to.